### ISSUE TITLE
Firefox: userEvent.click does not submit forms

### ISSUE THREAD

**Issue by anilanar** [testing-library/user-event#1075]
### Reproduction example

https://codesandbox.io/s/nervous-microservice-nmm3h7?file=/src/index.js

### Prerequisites

0. **Use Firefox, not jsdom**.
1. Render a form with an onSubmit handler and a button inside the form with a <button type="submit" />.
2. Use `userEvent.click` to click the button in the form.

### Expected behavior

Form's onSubmit handler should trigger.

### Actual behavior

Form's onSubmit handler does not trigger.

### User-event version

14.4.3

### Environment

Firefox 106.0.2 (64-bit) Linux

**Comment by anilanar** [comment]
I believe this bug affects more than it looks, also in chromiums, but I cannot prove it.

**Comment by anilanar** [comment]
Our [chromatic interaction tests](https://www.chromatic.com/docs/interactions) that runs in chrome (or chromium, not sure if headless or not), using storybook `play` functions, stopped working after upgrading from 14.2.1 to 14.4.3 due to clicks not triggering submit events for forms. Not exactly sure what version of chrome/chromium and I couldn't reproduce it locally with chrome/chromium, but could reproduce it with Firefox as you can see above.

Downgrading to 14.3.0 seems to fix the issue.

**Comment by ph-fritsche** [comment]
Thanks for reporting this.

Maybe this is somehow related to  #1002 
We're working on changing our test environment. Hopefully this will allow us to tackle these issues. See #1019

**Comment by anilanar** [comment]
@ph-fritsche This issue is introduced with 14.4.0 (one of #1003, #1007 or #1009?) and is not reproducible in 14.3.0, if that helps.

**Comment by mscrivo** [comment]
Seeing similar issues here with `userEvent.pointer`. We have a test that clicks and resizes and resizable drawer and the test was checking whether the size has changed, and it never does now with 14.4+, but worked fine on 14.3 and below.

**Comment by sergei-maertens** [comment]
I'm seeing this behaviour too but it's inconsistent - in some storybook tests `userEvent.click(...)` to submit a form works fine, for others I need to fall back on `fireEvent` and some other crutches to prevent my tests from asserting things too soon because my form submit handler is still executing :confused:

**Comment by kasperpeulen** [comment]
Workaround:

```ts
export const Submitting = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const submitButton = canvas.getByRole('button', { name: /Submit/i });
-   await userEvent.click(submitButton);
+   await fireEvent.click(submitButton);
  },
};
```

But `fireEvent` does a lot less than `userEvent`, so you might have to manually do the stuff that userEvent does (focus, hover etc).

@ph-fritsche I don't really understand why this works, because I thought userEvent uses fireEvent under the hood.

**Comment by nomoreyesterday** [comment]
```
export const EmptySubmit: Story = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const SubmitButton = canvas.getByRole('button', { name: /submit/i });
    await expect(SubmitButton).toBeInTheDocument();
    await fireEvent.click(SubmitButton)
    expect(canvas.getByText(/Username must be at least 2 characters./i)).toBeInTheDocument();
  },
```
![image](https://github.com/testing-library/user-event/assets/124575074/60aff866-92e8-4d48-91e7-104dd1b987b2)

fireEvent only didn't work for me but if add userEvent that's work

```
export const EmptySubmit: Story = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const SubmitButton = canvas.getByRole('button', { name: /submit/i });
    await expect(SubmitButton).toBeInTheDocument();
    await fireEvent.click(SubmitButton)
+ await userEvent.click(SubmitButton)
    expect(canvas.getByText(/Username must be at least 2 characters./i)).toBeInTheDocument();
  },
```
![image](https://github.com/testing-library/user-event/assets/124575074/b04c664a-1357-4e25-a583-a48f9d309c5f)

**Comment by yannbf** [comment]
@ph-fritsche is this something you're still considering working on? Thank you!

**Comment by ph-fritsche** [comment]
@yannbf Yes, I'd love User-Event to work in all major browsers. The [`test:toolbox` script](https://github.com/testing-library/user-event/blob/main/scripts/test.js) now allows to effectively compare our solutions in different environments. But fixing the differences between Jsdom and Chrome as well as working on general bug fixes/features will probably have priority over Firefox-specific issues.
